core/src/main/java/org/springframework/security/authentication/UsernamePasswordTypedAuthenticationToken.java

```diff
@@ -0,0 +1,85 @@
+package org.springframework.security.authentication;
+
+import org.springframework.security.core.SpringSecurityCoreVersion;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link AbstractUserDetailsAuthentication} implementation that is designed for simple
+ * presentation of a username and password.
+ * <p>
+ *
+ * @param <D> The Details Object the Token can map to.
+ * @author Peter Eastham
+ */
+public class UsernamePasswordTypedAuthenticationToken<D> extends AbstractUserDetailsAuthentication<String, D> {
+
+	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
+
+	private String credentials;
+
+	/**
+	 * Creates a token based on a supplied <tt>UserDetails</tt> instance and
+	 * <tt>String</tt> Object.
+	 *
+	 * @param principal a nonnull <tt>UserDetails</tt> for the principal.
+	 * @throws IllegalArgumentException if {@code principal.getAuthorities()} is null, or
+	 *                                  any Authorities in it are null.
+	 */
+	private UsernamePasswordTypedAuthenticationToken(UserDetails principal, String credentials, boolean authenticated) {
```

Comment:
Another thing to note. I was very torn on if this should be a public or private constructor. In a review of how `UsernamePasswordAuthenticationToken` is handled, I arrived to the conclusion the maximum visibility should be `protected`, however with no sub-classes or other reasons to expose it, I left it as private.